src/pages/workspaces/workspace/jobHistory/SubmissionDetails.js

```diff
@@ -0,0 +1,246 @@
+import * as Utils from 'src/libs/utils'
+import { wrapWorkspace } from 'src/pages/workspaces/workspace/WorkspaceContainer'
+
+
+const SubmissionDetails = _.flow(
+  forwardRef,
+  wrapWorkspace({
+    breadcrumbs: props => breadcrumbs.commonPaths.workspaceDashboard(props),
+    title: 'Job History', activeTab: 'job history'
+  }),
+  ajaxCaller
+)((props, ref) => {
+  const { namespace, name, submissionId, ajax: { Workspaces }, workspace: { workspace: { bucketName } } } = props
+
+  /*
+   * State setup
+   */
+  const [submission, setSubmission] = useState({})
+  const [methodAccessible, setMethodAccessible] = useState()
+  const [statusFilter, setStatusFilter] = useState([])
+  const [textFilter, setTextFilter] = useState('')
+  const [sort, setSort] = useState({ field: 'workflowEntity', direction: 'asc' })
+
+  /*
+   * Data fetchers
+   */
+  const getSubmission = Workspaces.workspace(namespace, name).submission(submissionId).get
+  const initialize = async () => {
+    try {
+      const submission = await getSubmission()
+      setSubmission(submission)
+
+      try {
+        const { methodConfigurationName: configName, methodConfigurationNamespace: configNamespace } = submission
+        await Workspaces.workspace(namespace, name).methodConfig(configNamespace, configName).get()
+        setMethodAccessible(true)
+      } catch {
+        setMethodAccessible(false)
+      }
+    } catch (e) {
+      reportError('Unable to fetch submission details', e)
+    }
+  }
+
+  Utils.useOnMount(() => { initialize() })
+  useEffect(() => {
+    if (_.some(({ status }) => collapseStatus(status) === 'running', submission.workflows)) {
+      const timeout = setTimeout(async () => {
+        try {
+          const submission = await getSubmission()
+          setSubmission(submission)
+        } catch (e) {
+          reportError('Unable to update submission details', e)
+        }
+      }, 60000)
+      return () => clearTimeout(timeout)
+    }
+  }, [getSubmission, submission])
+
+  /*
+   * Sub-component constructors
+   */
+  const makeStatusLine = (iconFn, text) => div({ style: { display: 'flex', marginTop: '0.5rem', fontSize: 14 } }, [
+    iconFn({ marginRight: '0.5rem' }), text
+  ])
+
+  const makeSection = (label, children) => div({ style: { flexBasis: '33%', marginBottom: '1rem' } }, [
+    div({ style: Style.elements.sectionHeader }, label),
+    h(Fragment, children)
+  ])
+
+  /*
+   * Data prep
+   */
+  const {
+    cost, methodConfigurationName: workflowName, methodConfigurationNamespace: workflowNamespace, submissionDate,
+    submissionEntity: { entityType, entityName } = {}, submitter, useCallCache, workflows = []
+  } = submission
+
+  const filteredWorkflows = _.flow(
+    _.filter(wf => {
+      if (_.isEmpty(statusFilter) || statusFilter.includes(wf.status)) {
+        const wfAsText = JSON.stringify(_.values(wf))
+        return _.every(term => Utils.textMatch(term, wfAsText), textFilter.split(/s+/))
```

Comment:
Should this regex be `/\s+/`?

Also, using `JSON.stringify` leads to some artifacts like `",` being a match. What about using `join` with a space?